roles/tobiko/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure podman is installed
+  become: true
+  ansible.builtin.package:
+    name: podman
+    state: present
+
+- name: Create tobiko directories
+  ansible.builtin.file:
+    path: "{{ cifmw_tobiko_artifacts_basedir }}"
+    state: directory
+
+- name: Create clouds.yaml
+  vars:
+    cifmw_tempest_artifacts_basedir: "{{ cifmw_tobiko_artifacts_basedir }}"
+  ansible.builtin.import_role:
+    name: tempest
+    tasks_from: create-clouds-file
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Copy CA bundle to cifmw_tobiko_artifacts_basedir
+  ansible.builtin.copy:
+    src: "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+    dest: "{{ cifmw_tobiko_artifacts_basedir }}"
+    mode: '0444'
+    owner: "{{ lookup('env', 'USER') }}"
+    group: "{{ lookup('env', 'USER') }}"
+    remote_src: true
+
+- name: Set proper permission for tobiko directory
+  ansible.builtin.command:
+    cmd: "podman unshare chown 42495:42495 -R {{ cifmw_tobiko_artifacts_basedir }}"
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Ensure we have tobiko container image
+  register: _tobiko_fetch_img
+  containers.podman.podman_image:
+    name: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+  retries: 5
+  delay: 5
+  until: _tobiko_fetch_img is success
+
+- name: Run tobiko
+  ignore_errors: true
+  containers.podman.podman_container:
+    name: tobiko
+    image: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+    state: started
+    auto_remove: "{{ cifmw_tobiko_remove_container }}"
```

Comment:
default value seems to be `false` for that auto_remove. Meaning we'll have that dangling container.
Since it defaults to `false`, it may be valuable to get a `cleanup.yml` that will take care of the various things, such as:
- container
- cloud.yaml generated for that specific role
- any other files that aren't logs?

Reply:
I am going to set both `cifmw_tobiko_remove_container` and `cifmw_tempest_remove_container` default values to true.
Regarding other cleanups, all the files that are created under `cifmw_tobiko_artifacts_basedir` look relevant to me:
- `clouds.yaml`
- `tobiko.conf`
- `podman_tobiko.log`
- and the tobiko results

Reply:
sounds good.